Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2003,19 +2048,32 @@ namespace Babylon
 
                     current.insert(inputSource.ID);
 
-                    auto found = m_idToInputSource.find(inputSource.ID);
-                    if (found == m_idToInputSource.end())
+                    auto inputSourceFound = m_idToInputSource.find(inputSource.ID);
+                    if (inputSourceFound == m_idToInputSource.end())
                     {
                         // Create the new input source, which will have the correct spaces associated with it.
```

Comment:
@satyapoojasama @rickfromwork does this mean we are always populating both hand information and game pad information for all OpenXR UWP form factors? What does the input source declare for vr headsets in terms of hands currently? What does hololens declare in terms of gamepad?

IIRC, hololens shouldn't support a gamepad. I don't think vr should support hands. But, right now it looks like everything is always populated. What are folks thoughts here?

Reply:
@chrisfromwork why shouldn't HoloLens support gamepad? It's possible to connect a wireless one via bluetooth and use it in other apps and the shell.

Reply:
Thinking longer-term, since this is technically OpenXR-based, it's conceivable to me that we'll end up in situations where we'll see novel combinations of input modalities and immersive vs. augmented rendering. Think Quest + Hands and HoloLens + controller (which, HoloLens 1 did ship with the clicker).

Ideally, we'd rely on the OpenXR runtime to tell us what is and isn't present and just expose the capabilities of that runtime.

Reply:
Side note on my comment - I keep forgetting gamepad here means a 6dof controller. I'm used to gamepad meaning xbox controller. I don't think we currently support 6dof on HoloLens (though I do remember talks in the past about wanting to).